src/main/java/com/eppo/sdk/helpers/RuleValidator.java

```diff
@@ -111,23 +111,45 @@ private static boolean matchesRule(
      */
     private static boolean evaluateCondition(
             SubjectAttributes subjectAttributes,
-            Condition condition
-    ) throws InvalidSubjectAttribute {
+            Condition condition) throws InvalidSubjectAttribute {
         if (subjectAttributes.containsKey(condition.attribute)) {
             EppoValue value = subjectAttributes.get(condition.attribute);
+            Optional<Version> valueSemVer = Version.tryParse(value.stringValue());
+            Optional<Version> conditionSemVer = Version.tryParse(condition.value.stringValue());
+
             try {
                 switch (condition.operator) {
                     case GTE:
-                        return Compare.compareNumber(value.doubleValue(), condition.value.doubleValue()
-                                , (a, b) -> a >= b);
+                        if (valueSemVer.isPresent() && conditionSemVer.isPresent()) {
+                            return valueSemVer.get().isHigherThanOrEquivalentTo(conditionSemVer.get());
+                        }
+
+                        return Compare.compareNumber(value.doubleValue(), condition.value.doubleValue(),
+                                (a, b) -> a >= b);
```

Comment:
Not you, but as it is written, I don't think we need `compareNumber()`.
```java
value.doubleValue() >= b condition.value.doubleValue();
```
_However,_ I think the spirit of this may have been to do numeric checks first. In that case we'd just pass the `EppoValue`s to `compareNumber()` and it would do the `isNumeric()` check on each.

_However,_ right now if either is not numeric the parsing will throw an exception, be caught by this outer try catch and return `false`, resulting in the same net outcome.

So my vote would still be to axe `compareNumber()` for now we can always add something like it back later should we want to.